### PR TITLE
More accurate error reporting when an Auth error occurs.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

**Comment by ehsannas** [comment]
Looks like ClangFormat made a couple of minor changes in this PR. I'm using `clang-format version 12.0.0`.

File: firestore/src/ios/credentials_provider_desktop.cc [hunk 1]
```diff
@@ -18,9 +19,35 @@ using auth::CredentialChangeListener;
 using auth::Token;
 using auth::TokenListener;
 using auth::User;
+using firebase::auth::AuthError;
 using util::Status;
 using util::StatusOr;
 
+Error ConvertAuthErrorToFirestoreError(int e) {
```
Comment (dconeybe) [inline firestore/src/ios/credentials_provider_desktop.cc:26]: Consider renaming `ConvertAuthErrorToFirestoreError` to `FirestoreErrorFromAuthError`.

IMO, this makes code at the call site more naturally readable:
```
Error firestoreError = ConvertAuthErrorToFirestoreError(code);
```
vs.
```
Error firestoreError = FirestoreErrorFromAuthError(code); 
```
See go/python-tips/054 for further rationale.

Comment (dconeybe) [inline firestore/src/ios/credentials_provider_desktop.cc:26]: nit: Please rename the argument `e` to a full English word. Non-standard abbreviations tend to detract from readabililty.

Comment (dconeybe) [inline firestore/src/ios/credentials_provider_desktop.cc:26]: Please add a comment to this function to document what the intended argument type is.

File: firestore/src/ios/credentials_provider_desktop.cc [hunk 2]
```diff
@@ -63,7 +90,7 @@ StatusOr<Token> ConvertToken(const Future<std::string>& future, App& app) {
     // is "from a different error domain".
     // TODO(b/174485290) Map `AuthError` values to Firestore `Error` values more
     // intelligently so as to enable retries when appropriate.
-    return Status(Error::kErrorUnknown,
+    return Status(ConvertAuthErrorToFirestoreError(future.error()),
```
Comment (dconeybe) [inline firestore/src/ios/credentials_provider_desktop.cc:93]: Please update the comment above this line.

**Review by dconeybe — CHANGES_REQUESTED** [review]

File: firestore/src/ios/credentials_provider_desktop.cc (hunk 1 above)
Comment (ehsannas) [reply firestore/src/ios/credentials_provider_desktop.cc]: Done

Comment (ehsannas) [reply firestore/src/ios/credentials_provider_desktop.cc]: Done

Comment (ehsannas) [reply firestore/src/ios/credentials_provider_desktop.cc]: Done

File: firestore/src/ios/credentials_provider_desktop.cc (hunk 2 above)
Comment (ehsannas) [reply firestore/src/ios/credentials_provider_desktop.cc]: Done.

File: firestore/src/ios/credentials_provider_desktop.cc [hunk 3]
```diff
@@ -18,9 +19,39 @@ using auth::CredentialChangeListener;
 using auth::Token;
 using auth::TokenListener;
 using auth::User;
+using firebase::auth::AuthError;
 using util::Status;
 using util::StatusOr;
 
+/**
+ * Takes an integer that represents an AuthError enum value, and returns a
```
Comment (dconeybe) [inline firestore/src/ios/credentials_provider_desktop.cc:27]: nit: Conventionally, we put backticks around type names. Please update this comment accordingly.

e.g. AuthError -> \`AuthError\`

**Review by dconeybe — DISMISSED** [review]

File: firestore/src/ios/credentials_provider_desktop.cc (hunk 3 above)
Comment (ehsannas) [reply firestore/src/ios/credentials_provider_desktop.cc]: Done

**Review by dconeybe — APPROVED** [review]

**Pull request merged**